DEVELOPER-GUIDE.MD

```diff
@@ -0,0 +1,25 @@
+[![Publish Release](https://github.com/ploigos/ploigos-step-runner/workflows/Publish%20Release/badge.svg)](https://github.com/ploigos/ploigos-step-runner/actions?query=workflow%3A%22Publish+Release%22)
```

Comment:
can you put the dev guide in a seperate merge request?

Reply:
Stashed for now.